src/github.com/cloudfoundry-incubator/webhooks/pkg/webhooks/manager/event.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/golang/glog"
+	"k8s.io/api/admission/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// Event stores the event details
+type Event struct {
+	AdmissionReview *v1beta1.AdmissionReview
+	crd             GenericResource
+	oldCrd          GenericResource
+}
+
+// NewEvent is a constructor for Event
+func NewEvent(ar *v1beta1.AdmissionReview) (*Event, error) {
+	crd, err := getGenericResource(ar.Request.Object.Raw)
+	if err != nil {
+		glog.Errorf("Could not get the GenericResource object %v", err)
+		return nil, err
+	}
+	oldCrd, err := getGenericResource(ar.Request.OldObject.Raw)
+	if err != nil {
+		glog.Errorf("Could not get the old GenericResource object %v", err)
+		return nil, err
+	}
+	crd.Status.lastOperation = getLastOperation(crd)
+	crd.Spec.options = getOptions(crd)
+	oldCrd.Status.lastOperation = getLastOperation(oldCrd)
+	oldCrd.Spec.options = getOptions(oldCrd)
+	crd.Status.appliedOptions = getAppliedOptions(crd)
+	oldCrd.Status.appliedOptions = getAppliedOptions(oldCrd)
+	return &Event{
+		AdmissionReview: ar,
+		crd:             crd,
+		oldCrd:          oldCrd,
+	}, nil
+}
+
+func (e *Event) isStateChanged() bool {
+	loNew := e.crd.Status.lastOperation
+	loOld := e.oldCrd.Status.lastOperation
+	glog.Infof("New: type: %s, state: %s\n", loNew.Type, loNew.State)
+	glog.Infof("Old: type: %s, state: %s\n", loOld.Type, loOld.State)
+	return loNew.Type == loOld.Type && loNew.State != loOld.State
+}
+
+func (e *Event) isPlanChanged() bool {
+	appliedOptionsNew := e.crd.Status.appliedOptions
+	appliedOptionsOld := e.oldCrd.Status.appliedOptions
+	return appliedOptionsNew.PlanID != appliedOptionsOld.PlanID
+}
+
+func (e *Event) isCreate() bool {
+	return e.crd.Status.lastOperation.Type == "create"
+}
+
+func (e *Event) isUpdate() bool {
+	return e.crd.Status.lastOperation.Type == "update"
+}
+
+func (e *Event) isSucceeded() bool {
+	return e.crd.Status.lastOperation.State == "succeeded"
+}
+
+func (e *Event) isMeteringEvent() bool {
+	if e.isStateChanged() && e.isSucceeded() {
+		if e.isUpdate() && e.isPlanChanged() {
+			return true
+		}
+		if e.isCreate() {
+			return true
+		}
+	}
```

Comment:
Probably you can simply write 

`
return (e.isUpdate() && e.isPlanChanged()) || e.isCreate()
`